### DISCUSSION TITLE
Decoupling library from integration version

### DISCUSSION THREAD

**Discussion by bmos** [andrew-codechimp/HA-Battery-Notes#464] [Ideas]
I'm not sure if this is worth the added complexity, but thought I'd bring it up:

It'd be quite possible/easy to download the library json from this or another repo within Home Assistant, which would allow rolling out new definitions continuously/more frequently than integration updates. This might be useful once development is slow/most work is on definitions.

If doing this, there would need to be some sort of schema version defined for the library so that if the library format changes the integration would know whether the library file used is compatible.

**Comment by andrew-codechimp** [comment]
This was actually added in version 1.2 a few weeks ago when the library really took off, it automatically downloads, schema version checks and replaces the library every 24 hours after HA has started.
The only thing it doesn't do is a discovery until HA restarts, which I noticed the other day and will probably be adding that next week.
